php/libraries/FileUploadValidator.class.inc

```diff
@@ -0,0 +1,241 @@
+<?php declare(strict_types=1);
+
+/**
+ * Class to upload files in a safe way.
+ *
+ * This class provides validation for files being uploaded to LORIS. It should
+ * serve as the ONLY way that files are uploaded so that all file uploading
+ * functionality can occur on a well-tested foundation.
+ *
+ * PHP Version 7
+ *
+ *  @category Main
+ *  @package  Behavioural
+ *  @author   John Saigle <john.saigle@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris
+ */
+
+/**
+ * Class to validate file uploads to LORIS.
+ *
+ *  @category Main
+ *  @package  Behavioural
+ *  @author   John Saigle <john.saigle@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris
+ */
+class FileUploadValidator
```

Comment:
`FileUploader` would be enough. The Validator part is implicit in the constructor code.
I would not expect a `FileUploadValidator` to have a upload function. (This is debatable)